Fixed database setup

Ahmad-Dimashkie opened the pull request:
Created the database connection to AWS rds database

Georges7Haddad commented on backend/src/database/database.py:
```diff
@@ -0,0 +1,29 @@
+from sqlmodel import Session, create_engine
+
+# Load the database_config.env file that is in the same directory as this file
+load_dotenv(
```
Please update the readme of the backend with instructions to set up your environment for others to follow

Georges7Haddad commented on backend/src/database/database.py:
```diff
@@ -0,0 +1,29 @@
+
+# A Session is what stores the objects in memory and keeps track of any changes
+# needed in the data then it uses the engine to communicate with the database.
+def get_session():
```
Can't you use this in your test?

Georges7Haddad commented on backend/src/tests/test_db_session.py:
```diff
@@ -0,0 +1,18 @@
+		with Session(engine) as session:
+			result = session.exec(text("SELECT 1"))
+			assert result.fetchone() is not None
+	except Exception as e:
```
There's no need for error handling in tests

Georges7Haddad commented on backend/src/tests/test_db_session.py:
```diff
@@ -0,0 +1,18 @@
+		pytest.fail(f"Database connection failed: {e}")
+
+
+if __name__ == "__main__":
```
There should be no need for that

Ahmad-Dimashkie commented:
addressed the comments. one question I had is: how do i share with you the config file (in the readme I said that I'll share it on google drive but is there another platform to share such files?)

Georges7Haddad commented:
Google drive is fine. Usually you would add collaborators on the cloud provider. That wouldn't be bad if you could do it so the others can get familiar with it as well

Ahmad-Dimashkie commented:
But for the connection (backend with Mysql workbench) they need a connection which requires a username and a password thats why I created a config file

Ahmad-Dimashkie commented:
Ohhh I didn't know I'll look into that thank you for the guidance

Georges7Haddad commented on backend/src/database/database.py:
```diff
@@ -0,0 +1,26 @@
+from sqlmodel import Session, SQLModel, create_engine
+
+load_dotenv(
+	dotenv_path=os.path.join(os.path.dirname(__file__), "database_config.env")
```
Once we deploy, we will need to remove this line. Cloud providers allow you to inject environment variables to your services so your service will read them directly from the environment instead of loading this "database_config.env" file to the environment and then reading from it. For now, you can keep it.

Ahmad-Dimashkie replied:
Okay noted thanks for the feedback, do we wait for another member of the group to review the changes and then we merge or do i merge now?

Georges7Haddad commented:
Please fix the CI check before merging. I think you will need to delete the test
